### DISCUSSION TITLE
Converting Flow MomentIDs to nbatopshot.com GUIDs

### DISCUSSION THREAD

**Discussion by alexdpoon** [rrrkren/topshot-explorer#14] [Q&A]
Hi, first of all, I just want to thank you sharing your Javascript code for interacting with the Flow blockchain for Top Shot. It's awesome. I love Top Shot but am new to Flow, am starting to think about making some analytic tools using Javascript.

Question for you: 

MomentIDs on the Flow blockchain are UInt64 (and PlayIDs and SetIDs are UInt32). But on the nbatopshot.com website, the link URLs use GUIDs, not integers.  For example, for the MomentID 806882, the nbatopshot.com GUID is d09c2c43-06c7-4303-8a3f-25c68b09e583, resulting in this [link](https://www.nbatopshot.com/moment/d09c2c43-06c7-4303-8a3f-25c68b09e583).  Do you know how I convert between the integer IDs and the GUIDs?

Thanks in advance.

**Comment by rrrkren** [comment]
Hi!
unfortunately the uuid is private data on topshot and won't be made public (for now at least). I'll let you know once that changes

**Reply by alexdpoon** [reply]
Hi, thank you for the reply! Yes let me know if this ever changes, especially since both the MomentID and the uuid are actually both currently public (MomentID is on the blockchain and the uuid is in the public-facing nbatopshot.com URL). It's just the linkage between the two that is opaque.

**Reply by Ng2k** [reply]
If the GUID is private, where does cryptoslam find it?
In [this](https://cryptoslam.io/nba-top-shot/mint/1902625) page there is the correct link to the moment.

Thanks for the time

**Reply by Nlitwinetz** [reply]
I am also curious how cryptoslam and others are getting these GUIDs. I suspect it is one of two ways:
1. They are scraping the TS website (which is a violation of the terms of use and they should get shut down).
2. They have access to the TS database where these GUIDs are stored.
